app/screens/edit_profile/components/form.test.tsx

```diff
@@ -436,4 +436,197 @@ describe('ProfileForm', () => {
         // Should render as a text field since no field definition exists
         expect(getByTestId('edit_profile_form.customAttributes.unknownField')).toBeTruthy();
     });
+
+    describe('SAML Field Disabling', () => {
+        it('should disable custom fields when SAML-linked', () => {
+            const customFields = [
+                TestHelper.fakeCustomProfileFieldModel({
+                    id: 'saml-field',
+                    name: 'SAML Field',
+                    type: 'text',
+                    attrs: {
+                        saml: 'Department', // SAML-linked
+                    },
+                }),
+                TestHelper.fakeCustomProfileFieldModel({
+                    id: 'normal-field',
+                    name: 'Normal Field',
+                    type: 'text',
+                    attrs: {
+                        saml: '', // Not SAML-linked
+                    },
+                }),
+            ];
+
+            const props = {
+                ...baseProps,
+                enableCustomAttributes: true,
+                customFields,
+                userInfo: {
+                    ...baseProps.userInfo,
+                    customAttributes: {
+                        'saml-field': {
+                            id: 'saml-field',
+                            name: 'SAML Field',
+                            type: 'text',
+                            value: 'Engineering',
+                        },
+                        'normal-field': {
+                            id: 'normal-field',
+                            name: 'Normal Field',
+                            type: 'text',
+                            value: 'Mobile Team',
+                        },
+                    },
+                },
+            };
```

Comment:
Rant: Lately, I am more inclined to use functions to create the base props, instead of objects. I don't like the neverending `{...foo, bar: {...foo.bar, ...}}`. If we were to put this as a function we would just do...
```
const props = getBaseProps();
props.enableCustomAttributes = true;
props.customFields = customFields;
props.userInfo.customAttributes = {...};
```

Which I feel it is easier to read. But yeah, doing this change here and now makes no sense, but wanted to rant 😜